pkg/storage/minio/lister.go

```diff
@@ -23,21 +36,15 @@ func (l *storageImpl) List(ctx context.Context, module string) ([]string, error)
 	if err != nil {
 		return nil, errors.E(op, err, errors.M(module))
 	}
+	ret := []string{}
 	for _, object := range objectCh.Contents {
 		if object.Err != nil {
 			return nil, errors.E(op, object.Err, errors.M(module))
 		}
-		parts := strings.Split(object.Key, "/")
-		ver := parts[len(parts)-2]
-		if _, ok := dict[ver]; !ok {
-			dict[ver] = struct{}{}
+		if ver, ok := l.getModuleSemVerFromKey(module, object.Key); ok {
```

Comment:
same as above, no need for this function. You can check that the path is correct by just making sure the module paths are equal. Therefore, let's keep the old code and add those 3 new lines: 

```
modpath := strings.Join(parts[:len(parts)-2], "/")
if modpath != module {
	continue
}
```

Reply:
basically agree, a bit different from the 3 new lines with checking go mod key only